modules/programs/vim.nix

```diff
@@ -60,6 +61,15 @@ let
         optional (isString p && hasAttr p vpkgs) vpkgs.${p};
   in concatMap getPkg cfg.plugins;
 
+  optionalPlugins = let
+    vpkgs = pkgs.vimPlugins;
+    getPkg = p:
+      if isDerivation p then
+        [ p ]
+      else
+        optional (isString p && hasAttr p vpkgs) vpkgs.${p};
```

Comment:
I don't think we need to support specifying packages as strings since it's been deprecated for regular plugins for some time now. Therefore no need to add changes to `warnings` and `packagesNotFound` too.